.github/workflows/test_with_clone.yaml

```diff
@@ -0,0 +1,275 @@
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT OWNERSHIP ON FUTURE DYNAMIC TABLES IN SCHEMA ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.SCHEMACHANGE TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+
+            - name: Give <CLONE>_PROXY_ADMIN ownership of all objects in synapse, synapse_raw, schemachange
+              shell: bash
+              run: |
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.synapse_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE} REVOKE CURRENT GRANTS;"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.synapse_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.synapse_raw_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE} REVOKE CURRENT GRANTS;"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.synapse_raw_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.schemachange_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE} REVOKE CURRENT GRANTS;"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.schemachange_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+
+            - name: Grant USAGE on <CLONE>_PROXY_ADMIN to SNOWFLAKE_CLONE_ADMIN_ROLE
+              shell: bash
+              run: |
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE} TO ROLE ${SNOWFLAKE_CLONE_ADMIN_ROLE};"
+
+            - name: Grant database-level privileges to SNOWFLAKE_CLONE_ADMIN_ROLE
+              shell: bash
+              run: |
+                # Grant database privileges to the clone admin. These are privileges that would have otherwise come with the ownership of the clone,
+                # but since the clone admin is acting as a pseudo-admin and does not directly own the clone, these need to be granted explicitly.
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT MODIFY ON DATABASE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED} TO ROLE ${SNOWFLAKE_CLONE_ADMIN_ROLE};"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT MONITOR ON DATABASE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED} TO ROLE ${SNOWFLAKE_CLONE_ADMIN_ROLE};"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT CREATE SCHEMA ON DATABASE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED} TO ROLE ${SNOWFLAKE_CLONE_ADMIN_ROLE};"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT CREATE DATABASE ROLE ON DATABASE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED} TO ROLE ${SNOWFLAKE_CLONE_ADMIN_ROLE};"
+
+
```

Comment:
I'm realizing that you've done a top-notch job of severing "admin" ties from the original admin/proxy admin accounts roles, but we don't yet have anything to sever ties from the analyst account role. This isn't a concern with the current behavior, since the SYNAPSE_DATA_WAREHOUSE_DEV_ANALYST role is an orphan as far as RBAC is concerned, but if we ever implement something like this for prod we'll want to be sure to revoke USAGE on each *ALL_ANALYST database role from the <DOMAIN>_ANALYST so that analysts won't be confused by all these `SYNAPSE_DATA_WAREHOUSE...` databases cluttering up their UI.

Reply:
Good point - they _would_ have visibility of the objects in the clone database since USAGE of the database itself is inherited by USAGE of the database roles. Yes, if we expand this pipeline, we can spin up some logic to programmatically retrieve the roles we need to handle privileges for instead of writing each one out explicitly.